lib/matplotlib/backends/backend_cairo.py

```diff
@@ -132,8 +132,18 @@ def _fill_and_stroke (self, ctx, fill_c, alpha, alpha_overrides):
             ctx.restore()
         ctx.stroke()
 
-    @staticmethod
-    def convert_path(ctx, path, transform):
+    def convert_path(self, ctx, path, transform):
+        # First clip the path to the actual rendering extents
+        points = path.get_extents().get_points()
+        width = points[1][0] - points[0][0]
+        height = points[1][1] - points[0][1]
+        wfactor = width / self.width
+        hfactor = height / self.height
```

Comment:
Not sure how to rescale these extents from display units to data units, any hint ?